src/agentscope/models/yi_model.py

```diff
@@ -0,0 +1,184 @@
+except ImportError:
+    openai = None
+
+
+class YiWrapperBase(ModelWrapperBase, ABC):
+    """The model wrapper for Yi API."""
+
+    def __init__(
+        self,
+        config_name: str,
+        model_name: str = None,
+        api_key: str = None,
+        region: str = "domestic",  # "domestic" or "overseas"
+        client_args: dict = None,
+        generate_args: dict = None,
+        budget: float = _DEFAULT_API_BUDGET,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the Yi client."""
+        if model_name is None:
+            model_name = config_name
+            logger.warning("model_name is not set, use config_name instead.")
+
+        super().__init__(config_name=config_name)
+
+        if openai is None:
+            raise ImportError(
+                "Cannot find openai package in current python environment.",
+            )
+
+        self.model_name = model_name
+        self.generate_args = generate_args or {}
+
+        base_url = (
+            "https://api.lingyiwanwu.com/v1"
+            if region == "domestic"
+            else "https://api.01.ai/v1"
+        )
+        self.base_url = base_url
+
+        if region == "overseas" and model_name not in ["yi-large"]:
+            logger.warning(
+                "Model %s may not be available for overseas region. "
+                "Only yi-large is confirmed to work. More information can be "
+                "found here https://platform.01.ai/docs#models-and-pricing",
+                model_name,
+            )
+        self.client = OpenAI(
+            api_key=api_key,
+            base_url=self.base_url,
+            **(client_args or {}),
+        )
+
+        # Set the max length of Yi model (this might need to be adjusted)
+        self.max_length = 4096  # Placeholder value, adjust as needed
+
+        # Set monitor accordingly
+        self._register_budget(model_name, budget)
+        self._register_default_metrics()
+
+    def _register_default_metrics(self) -> None:
+        # Set monitor accordingly
+        self.monitor.register(
+            self._metric("call_counter"),
+            metric_unit="times",
+        )
+        self.monitor.register(
+            self._metric("prompt_tokens"),
+            metric_unit="token",
+        )
+        self.monitor.register(
+            self._metric("completion_tokens"),
+            metric_unit="token",
+        )
+        self.monitor.register(
+            self._metric("total_tokens"),
+            metric_unit="token",
+        )
+
+    def format(
+        self,
+        *args: Union[Msg, Sequence[Msg]],
+    ) -> Union[List[dict], str]:
+        raise NotImplementedError(
+            f"Model Wrapper [{type(self).__name__}] doesn't "
+            f"implement the format method. Please implement it "
+            f"in the subclass.",
+        )
+
+
+class YiChatWrapper(YiWrapperBase):
+    """The model wrapper for Yi's chat API."""
+
+    model_type: str = "yi_chat"
+
+    def __call__(
```

Comment:
Please refer to `OpenAIChatWrapper` or `LiteLLMChatWrapper`, and add streaming mode of `YiChatWrapper `.